### PR TITLE
Workaround to show the files sidebar on Nextcloud 18

### PR DESCRIPTION
**Pull request by juliushaertl**

### CODE REVIEW

File: controller/editorcontroller.php [hunk 1]
```diff
@@ -670,13 +670,18 @@ public function config($fileId, $filePath = NULL, $shareToken = NULL, $inframe =
         }
 
         if ($folderLink !== NULL && $inframe !== 2) {
-            $params["editorConfig"]["customization"]["goback"] = [
-                "url"  => $folderLink
-            ];
+            list($majorVersion) = \OC_Util::getVersion();
+            if ($majorVersion < 18) {
```
Comment (LinneyS) [inline controller/editorcontroller.php]: For what?

File: js/main.js [hunk 2]
```diff
@@ -115,14 +115,34 @@
...
+                var btnShare = $('<a class="icon icon-menu-sidebar"></a>');
+                btnShare.on('click', function () {
+                    OCA.Files.Sidebar.file === "" ? OCA.Files.Sidebar.open(fileDir + '/' + fileName) : OCA.Files.Sidebar.close()
+                })
+                wrapper.prepend(btnShare)
+            }
+            wrapper.prepend(btnClose)
+            $('.header-right').append(wrapper)
+
+            if (typeof OCA.Files.Sidebar === 'undefined') {
```
Comment (LinneyS) [inline js/main.js]: I change the link regardless of the existence of the sidebar

File: controller/editorcontroller.php (hunk 1 above)
Comment (juliushaertl) [reply controller/editorcontroller.php]: The OCA.Files.Sidebar API is only available on Nextcloud 18, so the callback url only needs to be left out there.

File: js/main.js (hunk 2 above)
Comment (juliushaertl) [reply js/main.js]: The issue with keeping the link change is, that if the user reloads the page, the files app code whill not be loaded and therefore the sidebar will not work.

File: controller/editorcontroller.php (hunk 1 above)
Comment (LinneyS) [reply controller/editorcontroller.php]: I do not use `OCA.Files.Sidebar`

File: js/main.js (hunk 2 above)
Comment (LinneyS) [reply js/main.js]: The page with the editor does not allow to open the sidebar.
But having a link in the address bar is very important for reopening and copying the link

File: controller/editorcontroller.php (hunk 1 above)
Comment (juliushaertl) [reply controller/editorcontroller.php]: This is the new 18 API, as used in https://github.com/ONLYOFFICE/onlyoffice-nextcloud/pull/204/files#diff-77ab163850ef0fa98d6a9ee9fb6bbcafR128

File: js/main.js (hunk 2 above)
Comment (juliushaertl) [reply js/main.js]: Ok, i have an idea how to solve this then. I'll give that a try.

File: controller/editorcontroller.php (hunk 1 above)
Comment (LinneyS) [reply controller/editorcontroller.php]: Leave it for the current version? 
https://github.com/ONLYOFFICE/onlyoffice-nextcloud/blob/67a7b4b0c0cc8bbe5948c118084ed0b3acabf609/js/main.js#L132-L134
https://github.com/ONLYOFFICE/onlyoffice-nextcloud/blob/67a7b4b0c0cc8bbe5948c118084ed0b3acabf609/js/main.js#L158-L159

**Comment by juliushaertl** [comment]
Closing as discussed.

**Pull request closed without merging**